recite/admin.py

```diff
@@ -0,0 +1,143 @@
+import csv
+import io
+import json
+from zipfile import ZipFile
+
+from django import forms
+from django.contrib import admin
+
+from quran.models import Ayah
+
+from .models import Recitation, Reciter
+import os
+import tempfile
+import shutil
+from django.core.files import File
+
+
+class RecitationInline(admin.TabularInline):
+    """Inline tab of Recitations in admin page of Reciter"""
+    # Hide editing of single Recitation field
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
+    model = Recitation
+
+
+class ReciterForm(forms.ModelForm):
+    """Reciter Form has additional segments_file field of Recitations model"""
+    segments_file = forms.FileField()
+    audio_zip_file = forms.FileField()
+
+    class Meta:
+        model = Reciter
+        fields = '__all__'
+
+
+@admin.register(Reciter)
+class ReciterAdmin(admin.ModelAdmin):
+    """
+    Reciter admin page including Recitations
+    Modified save_model method allows to save Recitations
+    objects at the same model creation form
+    """
+    inlines = [
+        RecitationInline,
+    ]
+    form = ReciterForm
+
+    def save_model(self, request, obj, form, change):
+        super().save_model(request, obj, form, change)
+
+        # process csv file uploaded
+        segments_file = form.cleaned_data['segments_file'].file
+        csv_content = self.read_segments_csv(
+            segments_file, encoding=request.encoding)
+
+        # process audio zip file uploaded
+        audio_zip_file = form.cleaned_data['audio_zip_file']
+        temp_dir = self.get_unzipped_files_dir(audio_zip_file)
+        file_paths = self.get_file_paths(temp_dir)
+
+        # create Recitation objects from processed data
+        for row in csv_content:
+            segments_list = json.loads(row.get("segments"))
+            segments = ",".join(
+                (f"{segment[2]}:{segment[3]}" for segment in segments_list)
+            )
+
+            ayah = self.get_ayah_object_from_row(row)
+            file_ayah_path = self.get_file_audio_path(file_paths, ayah)
+
+            with open(file_ayah_path, 'rb') as f:
+                file_ayah = File(f)
+
+                recitation, _ = Recitation.objects.get_or_create(
+                    ayah=ayah,
+                    reciter=obj,
+                    segments=segments,
+                    audio=file_ayah
+                )
+
+        shutil.rmtree(temp_dir)
```

Comment:
If somewhere above an exception is thrown, then this line won't be executed and you will leave this dir in your FS. Use `tempfile.TemporaryDirectory` with context manager to avoid it. Here is an example: https://docs.python.org/3/library/tempfile.html#examples